precompiles/gov/tx.go

```diff
@@ -0,0 +1,54 @@
+// Copyright Tharsis Labs Ltd.(Evmos)
+// SPDX-License-Identifier:ENCL-1.0(https://github.com/evmos/evmos/blob/main/LICENSE)
+
+package gov
+
+import (
+	"fmt"
+
+	govkeeper "github.com/cosmos/cosmos-sdk/x/gov/keeper"
+
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	"github.com/ethereum/go-ethereum/common"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/evmos/evmos/v20/x/evm/core/vm"
+)
+
+const (
+	// VoteMethod defines the ABI method name for the gov Vote transaction.
+	VoteMethod = "vote"
+)
+
+// Vote claims the rewards accumulated by a delegator from multiple or all validators.
+func (p Precompile) Vote(
+	ctx sdk.Context,
+	origin common.Address,
+	contract *vm.Contract,
+	stateDB vm.StateDB,
+	method *abi.Method,
+	args []interface{},
+) ([]byte, error) {
+	msg, voterHexAddr, err := NewMsgVote(args)
+	if err != nil {
+		return nil, err
+	}
+
+	// If the contract is the voter, we don't need an origin check
+	// Otherwise check if the origin matches the delegator address
+	isContractVoter := contract.CallerAddress == voterHexAddr
```

Comment:
Do we need to add the auth transaction to enable the contract to send votes under the name of the user? Or the authz transaction is declared in another precompile?

Reply:
yes, we should add the authorization methods on this precompile to approve a certain address to perform actions on behalf of other. For now this is not supported on this precompile